docs/en/query_language/functions/array_functions.md

```diff
@@ -458,6 +458,193 @@ SELECT arraySlice([1, 2, NULL, 4, 5], 2, 3) AS res
 
 Array elements set to `NULL` are handled as normal values.
 
+## arraySort(\[func,\] arr, ...) {#array_functions-reverse-sort}
+
+Sorts the elements of the `arr` array in ascending order. If the `func` function is specified, sorting order is determined by the result of the `func` function applied to the elements of the array. If `func` accepts multiple arguments, the `arraySort` function is passed several arrays that the arguments of `func` will correspond to. Detailed examples are shown at the end of `arraySort` description.
+
+Example of integer values sorting:
+
+``` sql
+SELECT arraySort([1, 3, 3, 0]);
+```
+```
+┌─arraySort([1, 3, 3, 0])─┐
+│ [0,1,3,3]               │
+└─────────────────────────┘
+```
+
+Example of string values sorting:
+
+``` sql
+SELECT arraySort(['hello', 'world', '!']);
+```
+```
+┌─arraySort(['hello', 'world', '!'])─┐
+│ ['!','hello','world']              │
+└────────────────────────────────────┘
+```
+
+Consider the following sorting order for the `NULL`, `NaN` and `Inf` values:
+
+``` sql
+SELECT arraySort([1, nan, 2, NULL, 3, nan, -4, NULL, inf, -inf]);
+```
+```
+┌─arraySort([1, nan, 2, NULL, 3, nan, -4, NULL, inf, -inf])─┐
+│ [-inf,-4,1,2,3,inf,nan,nan,NULL,NULL]                     │
+└───────────────────────────────────────────────────────────┘
+```
+
+- `-Inf` values are first in the array.
+- `NULL` values are last in the array.
+- `NaN` values are right before `NULL`.
+- `Inf` values are right before `NaN`.
+
+Note that `arraySort` is a [higher-order function](higher_order_functions.md). You can pass a lambda function to it as the first argument. In this case, sorting order is determined by the result of the lambda function applied to the elements of the array.
+
+Let's consider the following example:
+
+``` sql
+SELECT arraySort((x) -> -x, [1, 2, 3]) as res;
+```
+```
+┌─res─────┐
+│ [3,2,1] │
+└─────────┘
+```
+
+In this example, for each element of the source array, the lambda function returns the "weight" of the element that will be applied for sorting, that is, [1 –> -1, 2 –> -2, 3 –> -3]. Since the elements are sorted by increasing of their "weight", the result is [3, 2, 1].
```

Comment:
* this example is basically arrayReverseSort, it'd better to mention this explicitly
* I think this is more often called "sorting key" instead of "weight"